src/main/webapp/go.js

```diff
@@ -0,0 +1,141 @@
+const FINAL_MSSG = 'Congrats, you\'ve finished the hunt!';
+
+const riddleArr = [];
+let destIndex;
+const destArr = [];
+
+window.onload = getHunt();
+
+/**
+ * Retrieves scavenger hunt data, and updates to the current destination
+ * to reflect the current state of the hunt.
+ */
+function getHunt() {
+  fetch(GO_URL).then((response) => response.json()).then((mssg) => {
+    destIndex = mssg.index;
+    for (let i = 0; i < mssg.items.length; i++) {
+      riddleArr.push(mssg.items[i].riddle.puzzle);
+      destArr.push(mssg.items[i].name);
+    }
+    updateToCurrentState(destIndex);
+  });
+}
+
+/**
+ * Updates the hunt to the current destination that the user is on.
+ *
+ * @param {int} index The index of the destination that the user needs
+ * to find.
+ */
+function updateToCurrentState(index) {
+  if (index >= 0) {
+    hideStartButton();
+    changeRiddleMessage(riddleArr[index]);
+    getDestName();
+  }
+  sendIndexToServlet(index);
+  toggleProceedButton(/* hide = */ true);
+}
+
+/**
+ * Retrieves the string the user submitted as the destination name.
+ */
+function getDestName() {
+  fetch(NAME_URL).then((response) => response.json()).then((mssg) => {
+    if (mssg === destArr[destIndex]) { // The user entered the correct name.
+      toggleProceedButton(/* hide = */ false);
+    }
+  });
+}
+
+/**
+ * Creates a new paragraph element from text.
+ *
+ * @param {String} text to be displayed on main page.
+ * @return {String} newLine New element to be appended to main page.
+ */
+function createLine(text) {
+  const newLine = document.createElement('p');
+  newLine.innerText = text;
+  return newLine;
+}
+
+/**
+ * Show or hide the proceed button.
+ *
+ * @param {boolean} hide Whether the proceed button should be hidden or shown.
+ */
+function toggleProceedButton(hide) {
+  const proceedButton = document.getElementById(PROCEED_ID);
+  if (hide) {
+    proceedButton.classList.add(INVISIBLE_CLASS);
+  } else {
+    proceedButton.classList.remove(INVISIBLE_CLASS);
+  }
+}
+
+/**
+ * Hide the start button.
+ */
+function hideStartButton() {
+  const startButton = document.getElementById(START_ID);
+  startButton.classList.add(INVISIBLE_CLASS);
+}
+
+/**
+ * Change the riddle text displayed on the mmain page.
+ * @param {String} text Text that the riddle should be changed to.
+ */
+function changeRiddleMessage(text) {
+  const riddle = document.getElementById(RIDDLE_ID);
+  riddle.innerHTML = '';
+  riddle.appendChild(createLine(text));
+}
+
+/**
+ * Update the scavenger hunt data with the current destination that
+ * the user is on.
+ * @param {int} index Index of the current destination the user needs to find.
+ */
+function sendIndexToServlet(index) {
+  const params = new URLSearchParams();
+  params.append(INDEX_PARAM, index);
+  fetch(GO_URL, {method: 'POST', body: params});
+}
+
+/**
+ * After the user correctly names the destination, proceed to
+ * the next destination in the hunt.
+ */
+function proceed() { //eslint-disable-line
```

Comment:
Could you add a comment to say why this is here? 'Cause it's used in an onclick attribute but eslint doesn't see that, right?